src/types/das-types.ts

```diff
@@ -0,0 +1,251 @@
+
+// getAssetsByOwner //
+export interface AssetsByOwner {
+  ownerAddress: string;
+  page: number;
+  limit?: number;
+  before?: string;
+  after?: string;
+  sortBy?: DASAssetSortingRequest;
+}
+
+// getAssetsByCreator //
+export type AssetsByCreator = {
+  creatorAddress: string;
+  page: number;
+  onlyVerified?: boolean;
+  limit?: number;
+  before?: string;
+  after?: string;
+  sortBy?: DASAssetSortingRequest;
+};
+// getAssetsByGroup //
+export type AssetsByGroup = {
+  groupValue: string;
+  groupKey: string;
+  page: number;
+  limit?: number;
+  before?: string;
+  after?: string;
+  sortBy?: DASAssetSortingRequest;
+};
+
+// searchAssets
+export interface SearchAssets {
+  page: number; // starts at 1
+  limit?: number;
+  before?: string;
+  after?: string;
+  creatorAddress?: string;
+  ownerAddress?: string;
+  jsonUri?: string;
+  grouping?: string;
+  burnt?: boolean;
+  sortBy?: DASAssetSortingRequest;
+  frozen?: boolean;
+  supplyMint?: string;
+  supply?: number;
+  interface?: string;
+  delegate?: number;
+  ownerType?: OwnershipModel;
+  royaltyAmount?: number;
+  royaltyTarget?: string;
+  royaltyTargetType?: RoyaltyModel;
+  compressible?: boolean;
+  compressed?: boolean;
+}
+
+// getAssetsByAuthority
+export type AssetsByAuthority = {
+  authorityAddress: string;
+  page: number;
+  limit?: number;
+  before?: string;
+  after?: string;
+  sortBy?: DASAssetSortingRequest;
+};
+// getAsset
+export type GetAsset = {
+  id: string;
+};
+// getAssetProof
+export type GetAssetProof = {
+  id: string;
+};
+// getSignaturesForAsset
+export type GetSignaturesForAsset = {
+  id: string;
+  page: number;
+  limit?: number;
+  before?: string;
+  after?: string;
+};
+
+// Sorting on response
+export interface DASAssetSorting {
+  sort_by: AssetSortBy;
+  sort_direction: AssetSortDirection;
+}
+// Sorting on request (camelCase)
+export type DASAssetSortingRequest = {
+  sortBy: AssetSortBy;
+  sortDirection: AssetSortDirection;
+};
+// Asset Response
+export type getAssetResponse = {
+  interface: DASInterface; // enum
+  id: string;
+  content: DASOption<DASContent>;
+  authorities: DASOption<DASAuthorities>;
+  compression: DASOption<DASCompression>;
+  grouping: DASOption<DASGrouping>;
+  royalty: DASOption<DASRoyalty>;
+  ownership: DASOwnership;
+  creators: DASOption<Array<DASCreators>>;
+  uses: DASOption<DASUses>;
+  supply: DASOption<DASSupply>;
+  mutable: boolean;
+  burnt: boolean;
+};
+
+export type getAssetResponseList = {
+  total: number;
+  limit: number;
+  page: number;
+  items: getAssetResponse[];
+};
+export interface getAssetProofResponse {
+  root: string;
+  proof: Array<string>;
+  node_index: number;
+  leaf: string;
+  tree_id: string;
+}
+export interface getSignatureResponse {
+  total: number;
+  limit: number;
+  page: DASOption<number>;
+  before: DASOption<string>;
+  after: DASOption<string>;
+  items: Array<Array<string>>;
+}
+
+// Ownership --
+export interface DASOwnership {
```

Comment:
Instead of prefixing everything each time with DAS, you can use a namespace (e.g. DAS.Ownership). This is good for types but don't use it for actual JS objects (prevents tree shaking for webapps)